x/concentrated-liquidity/types/keys.go

```diff
@@ -160,18 +165,24 @@ func KeyPositionId(positionId uint64) []byte {
 // Position Prefix Keys
 
 // KeyAddressPoolIdPositionId returns the full key needed to store the position id for given addr + pool id + position id combination.
+// WARNING: do not parse this index by splitting by key separator as it contains
+// raw address bytes in its structure.
 func KeyAddressPoolIdPositionId(addr sdk.AccAddress, poolId uint64, positionId uint64) []byte {
 	return []byte(fmt.Sprintf("%s%s%x%s%d%s%d", PositionPrefix, KeySeparator, addr.Bytes(), KeySeparator, poolId, KeySeparator, positionId))
 }
 
 // KeyAddressAndPoolId returns the prefix key used to create KeyAddressPoolIdPositionId, which only includes addr + pool id.
 // This key can be used to iterate over users positions for a specific pool.
+// WARNING: do not parse this index by splitting by key separator as it contains
+// raw address bytes in its structure.
 func KeyAddressAndPoolId(addr sdk.AccAddress, poolId uint64) []byte {
 	return []byte(fmt.Sprintf("%s%s%x%s%d", PositionPrefix, KeySeparator, addr.Bytes(), KeySeparator, poolId))
 }
 
 // KeyUserPositions returns the prefix key used to create KeyAddressPoolIdPositionId, which only includes the addr.
 // This key can be used to iterate over all positions that a specific address has.
+// WARNING: do not parse this index by splitting by key separator as it contains
+// raw address bytes in its structure.
```

Comment:
Wait are you sure this is raw bytes??? If so then we have problems, I thought this was hex encoded when I was going through it in #5470

Reply:
ref: https://github.com/osmosis-labs/osmosis/pull/5470#discussion_r1225693845

Reply:
What problems do we have? I thought that as long as we don't use a key separator for parsing keys, it should be fine

Reply:
You're right we encode hex: https://github.com/osmosis-labs/osmosis/pull/5470#discussion_r1225697435